### PR TITLE
Make pinned pressure and constraint phi0 a postprocessor and move to base class

### LINKED ISSUE #23745: Generalize and clean up FV constraints
## Reason

For coupling Navier Stokes module and/or Pronghorn with SAM, the following capability is needed:

1. Couple a postprocessor as value for `phi0` for constaints.
2. Pin face pressure in Navier-Stokes.

## Design

1. Move `phi0` from the child classes to the base class.
2. Make the parameter a postprocessor value.
3. Propagate change to Navier-Stokes action.
4. Add a class that does what `FVIntegralValueConstraint` does but for elements along sidesets. 

## Impact

Will not break anything

### PR DESCRIPTION
**Pull request by snschune**
## Reason

Closes #23745. This reduces some code duplicated in child classes and allows changing `phi0` during the simulation, and adds the ability to impose an average value on a boundary.

## Design

Move `phi0` to the base class and make it a postprocessor. Introduce new finite volume boundary constraints

## Impact

New constraint imposition capability along boundaries

### CODE REVIEW

File: framework/src/fvkernels/FVIntegralValueConstraint.C [hunk 1]
```diff
@@ -25,12 +25,11 @@ FVIntegralValueConstraint::validParams()
   InputParameters params = FVScalarLagrangeMultiplierConstraint::validParams();
   params.addClassDescription("This class is used to enforce integral of phi = volume * phi_0 "
                              "with a Lagrange multiplier approach.");
-  params.addParam<Real>("phi0", 0, "What we want the average value of the primal variable to be.");
```
Comment (GiudGiud) [inline framework/src/fvkernels/FVIntegralValueConstraint.C:28]: let s leave those lines in. They rename the parameter with a better docstring

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by lindsayad** [comment]
@snschune I'll do my modifications here and address @GiudGiud's review if that works for you

**Comment by lindsayad** [comment]
How do we want to add this to the action? We currently have `momentum_outlet_types` as well as `pin_pressure` and `pinned_pressure_type` parameters to play with

File: modules/navier_stokes/doc/content/source/fvbcs/INSFVAveragePressureValueBC.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# INSFVAveragePressureValueBC
+
+This object wraps [FVBoundaryIntegralValueConstraint.md], meaning that this
+object can be used to impose an average pressure value on a boundary. The
+`variable` parameter should correspond to the pressure variable. When applying a
+`INSFVAveragePressureValueBC` for the pressure, no `FVBCs` should be given for
+the velocity on the same `boundary`. In this way a zero viscous flux is
```
Comment (snschune) [inline modules/navier_stokes/doc/content/source/fvbcs/INSFVAveragePressureValueBC.md:7]: > When applying a `INSFVAveragePressureValueBC` for the pressure, no `FVBCs` should be given for
the velocity on the same `boundary`.

Is there something that makes this conceptually impossible? That is actually what I am trying to do in the 
approach that we discussed earlier.

File: framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# FVBoundaryIntegralValueConstraint
+
+This object implements the residuals that enforce the constraint
+
+!equation
+\int_{\partial \Omega} \phi = \int_{\partial \Omega} \phi_0
```
[the remark this answers is not in the document]
Comment (lindsayad) [reply framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md:6]: I've always taken `\partial \Omega` to mean the boundary of the domain, and that is the constraint I'm imposing: that the integral value over the boundary be equal to the integral of the desired value.

File: modules/navier_stokes/doc/content/source/fvbcs/INSFVAveragePressureValueBC.md (hunk 2 above)
Comment (lindsayad) [reply modules/navier_stokes/doc/content/source/fvbcs/INSFVAveragePressureValueBC.md:7]: When no FVBCs are applied for the velocity, then the momentum advection flux kernel runs on the boundary using data extrapolated from the interior. For the Euler equations (which obviously these are not), this type of boundary treatment is required: the eigenvalues for sub-Mach flow dictate that you have n-dim + 1 physical boundary conditions and one implicit boundary condition at an inflow boundary and one physical boundary condition and n-dim + 1 implicit boundary conditions at an outflow boundary.

But setting that aside... from the meeting you want to impose a `WCNSFVMomentumFluxBC` on the same boundary as this new `INSFVAveragePressureValueBC`? If that is the case, then yes, this is actually possible and will work with the code we have now. I can just amend this documentation

File: framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (snschune) [reply framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md:6]: That's what I would understand for `\partial \Omega` as well. The integral above is over the boundary, though, right? Therefore, the subscript of the integral could be changed to `\partial \Omega`.

[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md:6]: It's already `\partial \Omega` ?
just be explicit in the text below the equation what Omega and \partial \Omega are

[the remark this answers is not in the document]
Comment (snschune) [reply framework/doc/content/source/fvbcs/FVBoundaryIntegralValueConstraint.md:6]: Never mind. How was I convinced it wasn't??

File: framework/src/fvkernels/FVIntegralValueConstraint.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/fvkernels/FVIntegralValueConstraint.C:28]: setParamDescription ?
docstring isnt really used for parameters I think

Comment (GiudGiud) [reply framework/src/fvkernels/FVIntegralValueConstraint.C:28]: oh is this an existing routine?

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**